### PR TITLE
IOS-15606: part 3 to register background/foreground notification inst…

### PR DESCRIPTION
**Pull request by thuyBox**
…ead of resign/active

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @thuyBox has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m [hunk 1]
```diff
@@ -168,35 +168,44 @@ - (void)viewDidAppear:(BOOL)animated
                                      message:self.connectionErrorMessage];
     }
 
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(applicationWillResignActive:) name:UIApplicationWillResignActiveNotification object:nil];
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(applicationDidBecomeActive:) name:UIApplicationDidBecomeActiveNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(didReceiveBackgroundNotification:)
+                                                 name:UIApplicationDidEnterBackgroundNotification
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m:173]: This is not too late?

**Review by boxtcarpel — APPROVED** [review]
'will resign active' vs. 'did enter background'? How do we know that 'did enter background' isn't/won't be too late for the web view losing it's connection...

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m (hunk 1 above)
Comment (thuyBox) [reply BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m:173]: we don't get UIApplicationDidBecomeActiveNotification at all. Verified this works on iOS 11.3 device.

Comment (jlawton) [reply BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m:173]: Apple docs suggest it's okay. https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622997-applicationdidenterbackground?language=objc

**Pull request merged**